### ISSUE TITLE
cmplr.find_library() in Windows returns Linux style paths

### ISSUE THREAD

**Issue by adham-mellanox** [mesonbuild/meson#7005]
**Describe the bug**
When trying to link to a library in a Workdir that exists in a Network Drive, the generated library path is in Linux Style and cause the linker to fail with "unknown option".

**To Reproduce**
ext_deps += cc.find_library(<libname>.lib, dirs: meson.current_source_dir(), required: true) 
where current dir is a network path for example "\\10.0.0.1\mynetdir\workdir\"
the error in compilation after that with clang: unknown option "//10.0.0.1/mynetdir/workdir/<libname>.lib"
The meson.current_source_dir() return path in a good format, but when using find_library() the path being added has '/' instead of '\'.

**Expected behavior**
find_library will add Windows style path when it comes from network.
When moving the workdir to a local disk on the Windows everything works fine.

**system parameters**
Native windows 
meson: 0.53.2
Ninja: 1.9.0

**Comment by actondev** [comment]
Having the same issue with meson 0.60.0

**Comment by nirbheek** [comment]
Yes, we should at the very least detect network paths (starting with `\\`) and keep them as-is.
